apps/api/src/server/utils/mapping/map-key-dates.js

```diff
@@ -39,24 +39,24 @@ export const mapRequestToKeyDates = ({
 	return pick(allKeyDates, allKeyDateNames);
 };
 
+/**
+ * @param {Date} value
+ * @param {string} key
+ *
+ * @returns {string|number}
+ */
+const mapPreApplicationDates = (value, key) =>
+	// 'submissionAtPublished' is a string not a date, so we don't convert to a UNIX Timestamp
+	key === 'submissionAtPublished' ? `${value || ''}` : mapDateToUnixTimestamp(value);
+
 /**
  * @param {import('@pins/applications.api').Schema.ApplicationDetails} keyDates
  *
  * @returns {Object<string,Object<string, any>>} keyDateResponse
  */
 export const mapKeyDatesToResponse = (keyDates) => {
 	return {
-		preApplication: {
-			// 'submissionAtPublished' is a string not a date, so we don't convert to a UNIX Timestamp
-			...mapValues(
-				pick(
-					keyDates,
-					preApplicationDateNames.filter((key) => key != 'submissionAtPublished')
-				),
-				mapDateToUnixTimestamp
-			),
-			submissionAtPublished: keyDates.submissionAtPublished
-		},
+		preApplication: mapValues(pick(keyDates, preApplicationDateNames), mapPreApplicationDates),
```

Comment:
design expects properties to be in a specific order, at least in this section ("Anticipated submission date published" should be right before "Anticipated submission date internal").

in this way we only map the dates as it was before (so we exclude submissionAtPuslibhshed: string), but keeping the same order of the properties

Reply:
Makes sense, thanks for explanation! Also I think your code is more readable than mine anyway.